src/Worker.Extensions.DurableTask/HTTP/DurableHttpRequest.cs

```diff
@@ -35,6 +35,38 @@ public DurableHttpRequest(HttpMethod method, Uri uri, IDictionary<string, String
         this.AsynchronousPatternEnabled = asynchronousPatternEnabled;
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DurableHttpRequest"/> class with specified HTTP method, URI, headers, 
+    /// content, token source, and options to enable asynchronous pattern handling.
+    /// </summary>
+    /// <param name="method">Method used for HTTP request.</param>
+    /// <param name="uri">Uri used to make the HTTP request.</param>
+    /// <param name="headers">Headers added to the HTTP request.</param>
+    /// <param name="content">Content added to the body of the HTTP request.</param>
+    /// <param name="tokenSource">Token source for authentication.</param>
+    /// <param name="asynchronousPatternEnabled">Specifies whether the DurableHttpRequest should handle the asynchronous pattern.</param>
+    /// <param name="timeout">TimeSpan used for HTTP request timeout.</param>
+    /// <param name="httpRetryOptions">Retry options used for the HTTP request.</param>
+    public DurableHttpRequest(
+        HttpMethod method,
+        Uri uri,
+        IDictionary<string, StringValues>? headers = null,
+        string? content = null,
+        ManagedIdentityTokenSource? tokenSource = null,
```

Comment:
`ManagedIdentityTokenSource` is just one of many different types of token sources that we may want to support. Rather than accepting a specific concrete type, it would be better to accept an interface or abstract base type. That would allow us to support additional token source types in the future (workload identity credentials, Azure CLI credentials, etc.).